cylc/flow/network/scan.py

```diff
@@ -206,13 +233,14 @@ def _scan_subdirs(listing: List[Path], depth: int) -> None:
         for task in done:
             path, depth, contents = task.result()
             running.remove(task)
-            if dir_is_flow(contents):
+            is_flow = dir_is_flow(contents)
```

Comment:
Might be worth repeating the type annotation here so that nobody removes the `is_flow is False` check below
```suggestion
            is_flow: Optional[bool] = dir_is_flow(contents)
```